src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -139,7 +136,7 @@ export const DropdownMenu = <T,>({
         if (disabled) {
```

Comment:
```suggestion
        if (disabled && isPopupShown) {
```

Reply:
Why?

Reply:
if disabled and not isPopupShown then no toggle event

Reply:
useEffect was so much more simple